spec/data/node_spec.rb

```diff
@@ -275,6 +275,29 @@ def canonicalize(hw_info)
     end
   end
 
+  describe "matching on checkin" do
+    hw_id = "001122334455"
+
+    let (:tag) {
+      Tag.create(:name => "t1", :matcher => Razor::Matcher.new(["=", ["fact", "f1"], "a"]))
+    }
+    let (:node) {
+      Node.create(:hw_info => ["mac=#{hw_id}"], :facts => { "f1" => "a" })
+    }
+
+    it "should update tags if the node is marked installed" do
+      node.installed = 'test'
+      node.save
+      node.match_tags
+      tag.match?(node).should be_true
+
+      node.checkin('facts' => { 'f1' => 'a' })
```

Comment:
The main goal of this test is to verify that this call here is changing the tags on a node marked as installed. Can you add a check prior to this which verifies the previous state of the node's tags as different from the later state of the node's tags?